internal/db/inmemory/map.go

```diff
@@ -16,36 +16,64 @@ type Map[K comparable, V Value[K]] struct {
 }
 
 func (m *Map[K, V]) Add(k K, v V) {
-	//TODO implement me
-	panic("implement me")
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.data[k] = v
 }
 
 func (m *Map[K, V]) AddAll(vs []V) {
-	//TODO implement me
-	panic("implement me")
+	m.mu.Lock()
+	for _, v := range vs {
+		m.data[v.Key()] = v
+	}
+	m.mu.Unlock()
 }
 
 func (m *Map[K, V]) Remove(k K) {
-	//TODO implement me
-	panic("implement me")
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	if m.data != nil {
+		delete(m.data, k)
+	}
```

Comment:
Here you checked if m.data != nil, I suggest you create a helper function that automatically recreates the map when it's `nil` and access this everytime.

```go
func (m *Map[K, V]) safedata() map[K]V {
 if m.data == nil {
  m.data = make(map[K]V)
 }
 return m.data
}
```

Then everywhere you need access to the map, you call it using `m.safedata()`